code/modules/mob/living/carbon/give.dm

```diff
@@ -33,10 +33,18 @@
 					to_chat(usr, "<span class='warning'> You need to stay in reaching distance while giving an object.</span>")
 					to_chat(target, "<span class='warning'> [usr.name] moved too far away.</span>")
 					return
+				if((I.flags & NODROP) || (I.flags & ABSTRACT))
+					to_chat(usr, "<span class='warning'>\the [I.name] stays stuck to your hand when you try to give it!</span>")
+					to_chat(target, "<span class='warning'>\the [I.name] stays stuck to [usr.name]'s hand when you try to take it!</span>")
+					return
 				if((usr.hand && usr.l_hand != I) || (!usr.hand && usr.r_hand != I))
 					to_chat(usr, "<span class='warning'> You need to keep the item in your active hand.</span>")
 					to_chat(target, "<span class='warning'> [usr.name] seem to have given up on giving \the [I.name] to you.</span>")
 					return
+				if(target.resting)
```

Comment:
this should be an `incapacitated` check.

We don't want to give items to paralyzed people either.

Reply:
I'll see if I can refactor the code a bit more to make it new